src/Microsoft.AspNetCore.Mvc.Core/ApiBehaviorOptions.cs

```diff
@@ -52,5 +70,43 @@ public class ApiBehaviorOptions
         /// that are bound from form data.
         /// </summary>
         public bool SuppressConsumesConstraintForFormFileParameters { get; set; }
+
+
+        public bool AllowUseProblemDetailsForClientErrorResponses
+        {
+            get => _allowUseProblemDetailsForClientErrorResponses.Value;
+            set => _allowUseProblemDetailsForClientErrorResponses.Value = value;
+        }
+
+        public IDictionary<int, Func<ActionContext, ProblemDetails>> ProblemDetailsFactory { get; } = new Dictionary<int, Func<ActionContext, ProblemDetails>>
+        {
+            [400] = _ => new ProblemDetails
```

Comment:
We're also going to have to reckon with conventions + swagger + custom error type. Is this guidance for that case to just write a new convention?

Reply:
I was hoping we could update `DefaultApiConvention` ones to use `ProblemDetails`. Of course you write a new set of conventions if you modify this.